### ISSUE TITLE
AMD Vulkan API with texture display issues in RPCS3 emulator

### ISSUE THREAD

**Issue by vanderleivieirainacio** [GPUOpen-Drivers/AMDVLK#126]
Hello people. I found a while ago about the problem that has existed in the AMD Adrenalin Edition video driver Vulkan API since the release of this driver edition (Adrenalin). Edition) are lack of textures (eg water) and background distortions (such as textures overlapping each other causing a kind of ghost effect). The problem only happens in the Adrenalin Edition (Currently version 19.11.3) drivers and not in the Crimsom Relive Edition (latest version: 17.11.4) which works the texture display perfectly. 
I tested using the latest version of the RPCS3 emulator (v0.0.7-9186-5b9df53c) and the AMD Radeon Adrenalin Edition 19.11.3 and Crimson ReLive Edition 17.11.4 (2017) video drivers so I only changed the drivers for video and not the emulator.

![Anotação 2019-11-28 235128](https://user-images.githubusercontent.com/46516229/69906117-1ae17d80-139d-11ea-9a8f-687235709268.png)
AMD Vulkan API with texture and background distortion issues in RPCS3 emulator (Radeon Adrenalin Edition 19.11.3)

![Anotação 2019-11-28 223651](https://user-images.githubusercontent.com/46516229/69906137-67c55400-139d-11ea-89b2-ec67c4adedb9.png)
No texture problems or background distortions in the Vulkan API (Crimsom Relive Edition 17.11.4)

![Anotação 2019-11-28 235313](https://user-images.githubusercontent.com/46516229/69906171-f934c600-139d-11ea-9fd0-d2e568148aaf.png)
AMD Vulkan API with texture and background distortion issues in RPCS3 emulator (Radeon Adrenalin Edition 19.11.3)

![Anotação 2019-11-28 223947](https://user-images.githubusercontent.com/46516229/69906183-4a44ba00-139e-11ea-9121-b7eb97a19454.png)
No texture problems or background distortions in the Vulkan API (Crimsom Relive Edition 17.11.4)

![Anotação 2019-11-28 235220](https://user-images.githubusercontent.com/46516229/69906196-7bbd8580-139e-11ea-92d2-9822ca593cdb.png)
AMD Vulkan API with texture and background distortion issues in RPCS3 emulator (Radeon Adrenalin Edition 19.11.3)

![Anotação 2019-11-28 223838](https://user-images.githubusercontent.com/46516229/69906207-92fc7300-139e-11ea-97c9-bff2329584d4.png)
No texture problems or background distortions in the Vulkan API (Crimsom Relive Edition 17.11.4)

![Anotação 2019-11-29 000715](https://user-images.githubusercontent.com/46516229/69906213-bd4e3080-139e-11ea-9b55-dcc097a93c70.png)
AMD Vulkan API with texture and background distortion issues in RPCS3 emulator (Radeon Adrenalin Edition 19.11.3)

![Anotação 2019-11-28 225410](https://user-images.githubusercontent.com/46516229/69906245-6006af00-139f-11ea-8821-fc6119ba52b2.png)
No texture problems or background distortions in the Vulkan API (Crimsom Relive Edition 17.11.4)

![Anotação 2019-11-29 001725](https://user-images.githubusercontent.com/46516229/69906255-744aac00-139f-11ea-9c5b-94f0bd2c2df3.png)
AMD Vulkan API with texture and background distortion issues in RPCS3 emulator (Radeon Adrenalin Edition 19.11.3)

![Anotação 2019-11-28 230717](https://user-images.githubusercontent.com/46516229/69906265-96dcc500-139f-11ea-875e-eadb454c5ad0.png)
No texture problems or background distortions in the Vulkan API (Crimsom Relive Edition 17.11.4)

![Anotação 2019-11-29 001850](https://user-images.githubusercontent.com/46516229/69906268-b378fd00-139f-11ea-9f3e-f3eb0410e3ff.png)
AMD Vulkan API with texture and background distortion issues in RPCS3 emulator (Radeon Adrenalin Edition 19.11.3)

![Anotação 2019-11-28 230837](https://user-images.githubusercontent.com/46516229/69906274-c68bcd00-139f-11ea-8973-8a2dfcc98d05.png)
No texture problems or background distortions in the Vulkan API (Crimsom Relive Edition 17.11.4)

![Anotação 2019-11-29 002044](https://user-images.githubusercontent.com/46516229/69906293-0652b480-13a0-11ea-8ce5-c6b2aa60e9cd.png)
AMD Radeon Adrenalin Edition 19.11.3 video driver recently used for testing.

![Anotação 2019-11-28 230318](https://user-images.githubusercontent.com/46516229/69906313-6fd2c300-13a0-11ea-8ac5-ded9e2046c79.png)
AMD Radeon Crimson Relive Edition 17.11.4 (2017) video driver recently used for testing.

Note: NVIDIA video drivers currently work the same way as 2017 Crimson ReLive Edition 17.11.4. All AMD Crimson ReLive Edition video drivers edition Vulkan API works the textures perfectly. The Vulkan API problem only started with the new editions of AMD Adrenalin Edition video drivers.

**Comment by JacobHeAMD** [comment]
Hi @vanderleivieirainacio , could you please help to record it with vktrace so that we can debug it?

**Comment by vanderleivieirainacio** [comment]
Hi friend which program do i use to create problem trace, RadeonGPU Profile or Renderdoc? I need you to be patient with me to explain how I should use the programs to generate the vulkan trace.

Enviado do Email para Windows 10

De: Jacob He
Enviado:quinta-feira, 26 de dezembro de 2019 03:43
Para: GPUOpen-Drivers/AMDVLK
Cc:mastersonic; Mention
Assunto: Re: [GPUOpen-Drivers/AMDVLK] AMD Vulkan API with texture displayissues in RPCS3 emulator (#126)

Hi @vanderleivieirainacio , could you please help to record it with vktrace so that we can debug it?

**Comment by jinjianrong** [comment]
@vanderleivieirainacio please refer to below links to capture trace with Renderdoc.
https://renderdoc.org/docs/how/how_capture_frame.html
https://www.gamingonlinux.com/wiki/Using_Renderdoc_capture

**Comment by kd-11** [comment]
I'm fairly certain this issue was fixed on the emulator side. The cause of the difference is clamping of writes to gl_FragDepth in FS. Old drivers behaved like OpenGL and clamped the output so that values greater than 1 clamped and did not get clipped, new behavior does not do this. I could never find any explicit definitive statements about what is supposed to happen in such a situation, but all vulkan drivers seem to allow range extend and clipping behavior so we decided to just assume that is what spec requires.
